pkg/steps/template.go

```diff
@@ -421,18 +421,50 @@ func createOrRestartPod(podClient coreclientset.PodInterface, pod *coreapi.Pod)
 	return created, nil
 }
 
-func waitForPodDeletion(podClient coreclientset.PodInterface, name string, uid types.UID) error {
+func waitForTeardownToComplete(pod *coreapi.Pod) {
 	timeout := 300
 	for i := 0; i < timeout; i += 2 {
-		pod, err := podClient.Get(name, meta.GetOptions{})
-		if errors.IsNotFound(err) {
-			return nil
-		}
-		if err != nil {
-			return fmt.Errorf("could not retrieve deleting pod: %v", err)
+		if !podTeardownCompleted(pod) {
+			log.Printf("Waiting for pod %s to complete teardown ... (%ds/%d)", pod.GetName(), i, timeout)
+			time.Sleep(2 * time.Second)
+		} else {
+			break
 		}
-		if pod.UID != uid {
-			return nil
+	}
+}
+
+func getPodByName(podClient coreclientset.PodInterface, name string, uid types.UID) (*coreapi.Pod, error) {
+	pod, err := podClient.Get(name, meta.GetOptions{})
+	if errors.IsNotFound(err) {
+		return nil, nil
```

Comment:
I think returning `nil, nil` from a function called `getPodByName` when the pod is not found is quite confusing.

Reply:
Yeah, name is confusing, maybe `checkThatPodExists`?

Reply:
Renamed in 2852853